Public/Src/Cache/ContentStore/Distributed/NuCache/TransitioningContentLocationStore.cs

```diff
@@ -243,6 +243,19 @@ public Result<MachineLocation> GetRandomMachineLocation(IReadOnlyList<MachineLoc
             return _localLocationStore.ClusterState.GetRandomMachineLocation(except);
         }
 
+        /// <inheritdoc />
+        public bool IsMachineActive(MachineLocation machine)
+        {
+            if (_configuration.HasReadMode(ContentLocationMode.Redis))
+            {
+                return _redisContentLocationStore.IsMachineActive(machine);
+            }
+
+            return _localLocationStore.ClusterState.TryResolveMachineId(machine, out var machineId)

```

Comment:
Does it make sense to add a helper IsMachineActive to cluster state? (directly, or using extension method?)

Reply:
My usual way of thinking is: whether the change will make code more readable or not, but not whether we'll use the method in another place.

For instance, the following code is easier (IMO) to read because it deals with the same level of abstraction:

return _configuration.HashReadMode(Redis) 
  ?   _redisContentLocationStore.IsMachineActive(machine)
  : _localLocationStore.IsMachineActive(machine);

But the current version digs deeper into the cluster's state and is lower level.

---
In reply to: [329783222](https://github.com/microsoft/BuildXL/pull/963#discussion_r329783222) [](ancestors = 329783222)